src/test/java/com/productservice/services/ProductServiceTest.java

```diff
@@ -0,0 +1,104 @@
+package com.productservice.services;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.productservice.dao.entity.Product;
+import com.productservice.dao.repositorys.ProductRepository;
+import com.productservice.exceptions.ProductNotFoundException;
+import com.productservice.service.serviceimpl.ProductServicesImp;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class ProductServiceTest {
+
+	@Mock
+	private ProductRepository productRepository;
+
+	@InjectMocks
+	private ProductServicesImp productService;
+
+	static Product product1 = new Product();
```

Comment:
Field variables should be private(I'm assuming it's not accessed outside the class)